### ISSUE TITLE
TypeError: server.getConnections is not a function

### ISSUE THREAD

**Issue by sd65** [ldapjs/node-ldapjs#415]
Hi all,

Thanks for this nice npm package. 
I want to access the number of concurrent connections on the server.
So, I used `server.connections`.

It works but returns the message :
```
(node:21622) DeprecationWarning: Server.connections property is deprecated. Use Server.getConnections method instead.
```
So I tried to use `server.getConnections()` but :
```
Unhandled Rejection : TypeError TypeError: server.getConnections is not a function
```

I don't know how to fix this, I looked up the code and getConneections is supposed to work !
Is it a bug ?

**Comment by zelein** [comment]
I'm seeing this as well.

**Comment by jonekdahl** [comment]
@jsumners Let me see if I can tackle this.

As far as I can tell, `Server.getConnections(cb)` requires a callback so it cannot replace `Server.connections` directly since it is async. The options I can think of are:

 1. Mirror the Node API by removing Server.connection and adding Server.getConnections(cb) in the ldapjs API (simple, but a breaking change)
 1. Do some internal bookkeeping in ldapjs to keep track of the current connection count  (probably doable but it could get out of sync with the actual connection count if some corner case is not handled correctly)

Which option would be better? Are there any other alternatives?

**Comment by jsumners** [comment]
@jonekdahl option 1 sounds the best. I'd rather not try to keep track of things core is already tracking. Let them do the hard work.

Thank you for taking it on.

**Comment by jonekdahl** [comment]
@jsumners Given the fix in #598, I guess this issue can be closed?

**Comment by jsumners** [comment]
@jonekdahl yep. Thanks.
